### ISSUE TITLE
Multiple instances of karts using speed weighted animations

### ISSUE THREAD

**Issue by Benau** [supertuxkart/stk-code#2445]
It's mainly the speed weight object issue, let's take xue for example:

If you choose two xue in a game, only one propeller of xue will spin, and it depends
on the last world id of xue's speed only. It happens to ghost and real karts too.

When there's network play, I think we should fix this, as there should be no reason for blocking
network players to use the same karts.

**Comment by Benau** [comment]
```
    km->m_speed_weighted_objects.resize(m_speed_weighted_objects.size());
    for(size_t i=0; i<m_speed_weighted_objects.size(); i++)
    {
        // Master should not have any speed weighted nodes.
        assert(!m_speed_weighted_objects[i].m_node);
        km->m_speed_weighted_objects[i] = m_speed_weighted_objects[i];
        km->m_speed_weighted_objects[i].m_model =
            irr_driver->copyAnimatedMesh(m_speed_weighted_objects[i].m_model);
    }
```
seems to work...

**Comment by Benau** [comment]
Actually @hiker does the copyAnimatedMesh method for same karts issue in #1796, the situation here is quite similar to there.

So good idea to copy here too for speed weighted objects?

**Comment by Benau** [comment]
In https://github.com/supertuxkart/stk-code/tree/transparent_kart, I did copy for wheels and speed weighted objects for the transparency, as it seems to be a simpler solution to identify ghost and non-ghost...

**Comment by Benau** [comment]
In my fix_2445 branch, I make vao manager possible to create a new vao for each same mesh buffer from different animated node, so it should be possible to render same animated model with copying them in instanced rendering.

@deveee, do you think you can try it on your android too? It can save some memory usage in my testing .

**Comment by Benau** [comment]
Branch merged

**Comment by deveee** [comment]
I didn't look at your patch that carefully, but AFAIK we are using
instanced/indirect rendering when glDrawArraysIndirect is available (by
default in OpenGL 4.0 and GLES 3.1). I disabled the code that is not
available in GLES 3.0 to make it working on more devices. It is possible to
dynamically load these functions when this extension is available, so maybe
it will be done in future.

I can look at your branch when I'll find some time.

**Comment by Benau** [comment]
Thx
